### PR TITLE
Fix most of the test failures

### PR DESCRIPTION
**Pull request by remi-stripe**
This PR tries to fix most of the tests failures in the library. All XUnit tests were fixed (at least with my API keys). Only fixed one of the other tests. Most tests complain about file uploads (I think) but I could not figure out what the issue is yet.

r? @brandur-stripe

### CODE REVIEW

File: src/Stripe.Tests.XUnit/accounts/_fixture.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@ public account_fixture()
                         new StripeAccountAdditionalOwner
                         {
                             FirstName = "Bumble", LastName = "B",
-                            BirthDay = 29, BirthMonth = 8, BirthYear = 2013,
+                            BirthDay = 29, BirthMonth = 8, BirthYear = 1980,
```
Comment (brandur-stripe) [inline src/Stripe.Tests.XUnit/accounts/_fixture.cs]: I guess this was broken by the age check that Security recently put in? Might be worth adding a comment to the effect that the `BirthYear` is somewhat important.

File: src/Stripe.Tests.XUnit/plans/_cache.cs [hunk 2]
```diff
@@ -23,7 +23,7 @@ public static StripePlanCreateOptions GetPlanCreateOptions(string planName = "pl
                 Amount = 1000,
                 Currency = "usd",
                 Name = Guid.NewGuid().ToString(),
-                Id = Guid.NewGuid().ToString(),
+                Id = Guid.NewGuid().ToString() + DateTime.UtcNow.Ticks,
```
Comment (brandur-stripe) [inline src/Stripe.Tests.XUnit/plans/_cache.cs]: Any idea why this works? A GUID should already be universally unique (unless some crazy stubbing has been applied).

**Comment by brandur-stripe** [comment]
A couple minor comments, but looking good. Thanks for taking the initiative here Remi!

File: src/Stripe.Tests.XUnit/plans/_cache.cs (hunk 2 above)
Comment (remi-stripe) [reply src/Stripe.Tests.XUnit/plans/_cache.cs]: @brandur-stripe well I think I'm just unlucky enough? I keep getting "plan already exists" over and over when running the test suite and those conflict on the random ids. But I have ran the test suites hundreds of times so maybe it's due to that?

File: src/Stripe.Tests.XUnit/accounts/_fixture.cs (hunk 1 above)
Comment (remi-stripe) [reply src/Stripe.Tests.XUnit/accounts/_fixture.cs]: I'm happy to put a comment though it feels self explanatory that the DOB should be a valid one and not for a 4 year old. Let me know if it's worth the fix though

File: src/Stripe.Tests.XUnit/plans/_cache.cs (hunk 2 above)
Comment (brandur-stripe) [reply src/Stripe.Tests.XUnit/plans/_cache.cs]: Just for general interest, I looked it up and .NET uses a V4 UUID generator (random).

Here's a [good writeup from a Quora answer](https://www.quora.com/Has-there-ever-been-a-UUID-collision):

> UUIDs have 122 bits of entropy so the chance of two random UUIDs colliding is about 10^-37.
>
> If you generate 2^46 UUIDs (approximately 1 petabyte of entropy) the chance of getting a collision is 1 in 50 billion.[1]  You're 170 times more likely to win the jackpot in Powerball.[2]

Meaning to say that the chances of generating a duplicate GUID (modulo a bug in the underlying software) are ~zero. I guess it's not a big deal, but I think that the required addition of extra entropy to this string probably means that we have some other bug or feature that we don't completely understand somewhere.

File: src/Stripe.Tests.XUnit/accounts/_fixture.cs (hunk 1 above)
Comment (brandur-stripe) [reply src/Stripe.Tests.XUnit/accounts/_fixture.cs]: Alright, fair enough.

File: src/Stripe.Tests.XUnit/plans/_cache.cs (hunk 2 above)
Comment (remi-stripe) [reply src/Stripe.Tests.XUnit/plans/_cache.cs]: I think you're right though I'm not grasping why I experience this. My ids are definitely GUIDs and I can reproduce reliably. Unless I add some logs that is, when I do the error does not happen anymore.
I guess this is the wrong fix and we should understand what's causing this issue in the first place.

Comment (remi-stripe) [reply src/Stripe.Tests.XUnit/plans/_cache.cs]: Okay, I think the issue here is that we go into `GetPlan()` and the plan does not exist yet. But when we go into `GetPlanCreateOptions()` another concurrent call to `GetPlanCreateOptions()` already initialized the options in the `Items` array. This causes another part of the code to try and create a plan a second time with the same option.

I do not understand why adding a timestamp fix those. It's likely a side-effect and not a solution per se.

I guess the right solution is to take a lock in `GetPlan()` and release it only when leaving.

Comment (remi-stripe) [reply src/Stripe.Tests.XUnit/plans/_cache.cs]: removed this attempt for now as it's the wrong approach.

**Comment by remi-stripe** [comment]
@brandur-stripe PTAL. I changed the fix for the subscriptions' tests failures.

**Comment by brandur-stripe** [comment]
Way better!

LGTM.

**Comment by remi-stripe** [comment]
@brandur-stripe @fred-stripe I wonder if we should merge this one? It's not a change to the library itself so it can be merged without needing a release right?
@gh-stripe is working on another modification and encountered all those tests errors too

**Comment by brandur-stripe** [comment]
Yeah, I think we should. And BTW, after seeing Andrew's instructions, I'm fairly confident that we could ship a release without too much trouble, so we should try that at some point.

**Pull request merged**